src/librustc_lint/builtin.rs

```diff
@@ -1952,3 +1952,171 @@ impl EarlyLintPass for Async2018 {
         }
     }
 }
+
+
+pub struct ExplicitOutlivesRequirements;
+
+impl LintPass for ExplicitOutlivesRequirements {
+    fn get_lints(&self) -> LintArray {
+        lint_array![EXPLICIT_OUTLIVES_REQUIREMENTS]
+    }
+}
+
+impl ExplicitOutlivesRequirements {
+    fn collect_outlives_bound_spans(&self, bounds: &hir::GenericBounds, infer_static: bool) -> Vec<(usize, Span)> {
+        let mut bound_spans = Vec::new();
+        for (i, bound) in bounds.iter().enumerate() {
+            if let hir::GenericBound::Outlives(lifetime) = bound {
+                let is_static = match lifetime.name {
+                    hir::LifetimeName::Static => true,
+                    _ => false
+                };
+                if is_static && !infer_static {
+                    // infer-outlives for 'static is still feature-gated (tracking issue #44493)
+                    continue;
+                }
+                bound_spans.push((i, bound.span()));
```

Comment:
So this just removes **all** outlives requirements -- but I think there are still cases where outlives declarations are required, though they are somewhat obscure. You can find an example in the ["where outlives requirements are still required" section of the RFC](https://rust-lang.github.io/rfcs/2093-infer-outlives.html#where-explicit-annotations-would-still-be-required). 

Probably the *right* way to do this would be to actually run the [`inferred_outlives_of`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc/ty/struct.TyCtxt.html#method.inferred_outlives_of) query and check whether this bound appears in the result. If so, it can be removed.

Reply:
Thanks for the pointer! :sparkling_heart: